lib/importproject.cpp

```diff
@@ -533,45 +534,182 @@ namespace {
+					auto right = parseString(i);
+					replaceAll(left, "$(Configuration)", p.configuration);
+					replaceAll(left, "$(Platform)", p.platformStr);
+					replaceAll(right, "$(Configuration)", p.configuration);
+					replaceAll(right, "$(Platform)", p.platformStr);
+					initialized = true;
+					currentVal = equal ? left == right : left != right;
+					continue;
+				}
+				if (startsHere(i, "$(Configuration."))
+				{
+					initialized = true;
+					if (startsHere(i, "Contains("))
+					{
+						auto contains = parseString(i);
+						currentVal = p.configuration.find(contains) != std::string::npos;
+					}
+					else if (startsHere(i, "StartsWith("))
+					{
+						auto startsWith = parseString(i);
+						currentVal = p.configuration.find(startsWith) == 0;
+					}
+					else if (startsHere(i, "EndsWith("))
+					{
+						auto endsWith = parseString(i);
+						currentVal = p.configuration.rfind(endsWith) == p.configuration.length() - endsWith.length();
+					}
+					else
+					{
+						throw std::runtime_error("Unexpected function call!");
+					}
+					skipSpaces(i);
+					if (!startsHere(i, "))"))
+					{
+						throw std::runtime_error("Expecting end of expression!");
+					}
+					continue;
+				}
+				throw std::runtime_error("Unhandled expression!");
+			}
+			if (!initialized)
+			{
+				throw std::runtime_error("Expected expression here!");
+			}
+			return currentVal;
+		}
+
         // see https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-conditions
         // properties are .NET String objects and you can call any of its members on them
         bool conditionIsTrue(const ProjectConfiguration &p) const {
-            if (mCondition.empty())
-                return true;
-            std::string c = '(' + mCondition + ");";
-            replaceAll(c, "$(Configuration)", p.configuration);
-            replaceAll(c, "$(Platform)", p.platformStr);
-
-            // TODO: improve evaluation
-            const Settings s;
-            TokenList tokenlist(s, Standards::Language::C);
-            tokenlist.createTokensFromBuffer(c.data(), c.size()); // TODO: check result
-            // TODO: put in a helper
-            // generate links
-            {
-                std::stack<Token*> lpar;
-                for (Token* tok2 = tokenlist.front(); tok2; tok2 = tok2->next()) {
-                    if (tok2->str() == "(")
-                        lpar.push(tok2);
-                    else if (tok2->str() == ")") {
-                        if (lpar.empty())
-                            break;
-                        Token::createMutualLinks(lpar.top(), tok2);
-                        lpar.pop();
-                    }
-                }
-            }
-            tokenlist.createAst();
-            for (const Token *tok = tokenlist.front(); tok; tok = tok->next()) {
-                if (tok->str() == "(" && tok->astOperand1() && tok->astOperand2()) {
-                    // TODO: this is wrong - it is Contains() not Equals()
```

Comment:
We should have filed a ticket for this since fixing it is a behavior change. Are you able to do that or should I file one?